submissions/antonkottans/tjsw-post-oop/index.js

```diff
@@ -0,0 +1,163 @@
+class Creature {
+    constructor({ name, gender, friends = [] }) {
+        this.name = name;
+        this.gender = gender;
+        this.friends = friends;
+        this.keys = ["name", "gender", "friends"];
+    }
+
+    returnFullInfo() {
+        return this.keys
+            .map((key) => {
+                if (this[key]) {
+                    if (key === "friends")
+                        if (this["friends"].length != 0)
+                            return (
+                                "friends:" +
+                                this["friends"]
+                                    .map((friend) => friend.name)
+                                    .join(", ")
+                            );
+                        else return null;
+                    else return key + ":" + this[key];
+                } else return null;
+            })
```

Comment:
Can you think of any way to simplify logic here?
As a peer developer I want methods be simpler and less nested.
Also each class should handle its own properties only.

You may benefit from methods overloading and there you won't need a property to store "keys" (which is BTW quite generic name per se).

Check the materials on OOP/SOLID in the original project's docs.